### PR TITLE
pass in stopCh to cloud provider Initialize method for custom controllers

### PR DESCRIPTION
**Pull request by andrewsykim**
**What type of PR is this?**
/kind api-change
/kind cleanup

**What this PR does / why we need it**:
Passes stopCh which can be passed into any custom controllers specific to a cloud provider (GCE clusterID watcher being one example). 

**Special notes for your reviewer**:
We haven't had a discussion around if this "api change" (the interface being the "api") should be done so happy to discuss that here (or wait for the next SIG Cloud Provider meeting). My take here is that we should make obvious/trivial changes now while the cloud provider interface is still "beta". The downside being out-of-tree providers have to update those methods when they vendor in Kubernetes 1.13, but I think it's worth the trade off (speaking as someone who maintains an out-of-tree provider). 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

cc @kubernetes/sig-cloud-provider 

/assign @cheftako @dims

### CODE REVIEW

**Comment by dims** [comment]
@andrewsykim tough question to answer, do we expect any more changes? :)

**Comment by andrewsykim** [comment]
I think we first need to answer the question of whether we should treat interfaces like APIs in the first place. We sort of spoke about this in our last SIG meeting but there were questions raised about if method signature changes count as "breaking changes". IMO we should focus less about method signatures changes and more about behaviour changes. If we did treat like like an API, maybe we should start versioning. Anyways, we had agreed to come to a solid agreement on this by KubeCON Seattle :P. 

Related to this PR, I'm not in a rush to merge this in so I'm okay to wait til we reach some sort of consensus, it just seems like the right code change based on what `Initialize` was originally designed for.

File: pkg/cloudprovider/providers/gce/gce_clusterid.go [hunk 1]
```diff
@@ -105,7 +105,7 @@ func (gce *GCECloud) watchClusterID() {
 	var controller cache.Controller
 	gce.ClusterID.store, controller = cache.NewInformer(newSingleObjectListerWatcher(listerWatcher, UIDConfigMapName), &v1.ConfigMap{}, updateFuncFrequency, mapEventHandler)
 
-	controller.Run(nil)
+	controller.Run(stop)
```
Comment (cheftako) [inline pkg/cloudprovider/providers/gce/gce_clusterid.go]: Wondering if it would be good to this in a separate PR. I believe we have other such fixes which need to be made. Might be better to not conflate the Interface change with actual fixes which may have unexpected side effects.

**Review by cheftako — APPROVED** [review]
/lgtm

File: pkg/cloudprovider/providers/gce/gce_clusterid.go (hunk 1 above)
Comment (andrewsykim) [reply pkg/cloudprovider/providers/gce/gce_clusterid.go]: I can at least do separate commit

Comment (andrewsykim) [reply pkg/cloudprovider/providers/gce/gce_clusterid.go]: done, PTAL :)

**Review by cheftako — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/cloud-provider/cloud.go [hunk 2]
```diff
@@ -42,8 +42,8 @@ type ControllerClientBuilder interface {
 // Interface is an abstract, pluggable interface for cloud providers.
 type Interface interface {
 	// Initialize provides the cloud with a kubernetes client builder and may spawn goroutines
-	// to perform housekeeping activities within the cloud provider.
-	Initialize(clientBuilder ControllerClientBuilder)
+	// to perform housekeeping or run custom controllers specific to the cloud provider
```
Comment (lavalamp) [inline staging/src/k8s.io/cloud-provider/cloud.go]: Document the semantics of the stop channel. (e.g. is it to tell the cloud provider to stop, or to find out when it stops?)

File: staging/src/k8s.io/cloud-provider/cloud.go [hunk 3]
```diff
@@ -42,8 +42,8 @@ type ControllerClientBuilder interface {
...
+	Initialize(clientBuilder ControllerClientBuilder, stop <-chan struct{})
```
Comment (lavalamp) [inline staging/src/k8s.io/cloud-provider/cloud.go]: This breaks go compatibility, is that OK? I think it won't be OK once there's out of tree implementors?

Comment (andrewsykim) [reply staging/src/k8s.io/cloud-provider/cloud.go]: This is a discussion we (SIG Cloud Provider) are having right now. There are some differing opinions around if breaking the interface is okay as long as the behaviour is not changed. In my opinion at least, this is fine because the SIG that defines the interface works closely with the implementors and we can form consensus on breaking changes like this one before moving forward. But you are correct in that we should have a definitive answer for this when we have stable out-of-tree providers. For now we are incrementally trying to put the interface at a better spot while we are still "beta".

File: staging/src/k8s.io/cloud-provider/cloud.go (hunk 2 above)
Comment (andrewsykim) [reply staging/src/k8s.io/cloud-provider/cloud.go]: done
